### DISCUSSION TITLE
Trying to use USB data port on RPi zero 2 w

### DISCUSSION THREAD

**Discussion by jlodge10** [mikebrady/shairport-sync#1791] [Q&A]
I want to set it up to plug into the USB A port in my car that it recognizes as an "iPod" input and provides the necessary power. However, the only option for audio output I can get is for the HDMI. I followed all of the directions on here step by step but when I plug it into the port and run raspi-config I only have the HDMI output in the audio settings and when I run cat /proc/asound/modules I only have the 0 vc4 which is the HDMI port. Any ideas how to fix this? Also, will this setup allow the retention of steering wheel controls?

I attached kernel logs if that would be helpful.
```
dmesg | grep usb
[    0.130257] usbcore: registered new interface driver usbfs
[    0.130338] usbcore: registered new interface driver hub
[    0.130407] usbcore: registered new device driver usb
[    0.130641] usb_phy_generic phy: supply vcc not found, using dummy regulator
[    0.130825] usb_phy_generic phy: dummy supplies not allowed for exclusive requests
[    2.198613] usbcore: registered new interface driver lan78xx
[    2.201274] usbcore: registered new interface driver smsc95xx
[    3.186427] dwc_otg 3f980000.usb: DWC OTG Controller
[    3.188595] dwc_otg 3f980000.usb: new USB bus registered, assigned bus number 1
[    3.190794] dwc_otg 3f980000.usb: irq 89, io mem 0x00000000
[    3.197330] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.01
[    3.201606] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.203867] usb usb1: Product: DWC OTG Controller
[    3.206091] usb usb1: Manufacturer: Linux 6.1.21-v7+ dwc_otg_hcd
[    3.208352] usb usb1: SerialNumber: 3f980000.usb
[    3.216538] usbcore: registered new interface driver usb-storage
[    3.236101] usbcore: registered new interface driver usbhid
[    3.238388] usbhid: USB HID core driver
[    9.307438] usbcore: registered new interface driver brcmfmac
```

**Comment by mikebrady** [comment]
Thanks for the post. I'm afraid that I may disappoint you, but let me see if I can explain the situation, which is based on a guess:

I am guessing that your car stereo is capable of playing audio from an iPad or an iPhone that is connected over a USB-A-to-Lightning (or USB-A-to-USB-C) cable. In addition, I guess that you can see the audio metadata (e.g. track name, cover art) on the car audio screen and you jump to the next/previous track, etc. (From experience, I know that some Fiats and other Stellantis products do this...)
Anyway, if my guess is right, then I'm afraid that Shairport Sync does not have a role; it doesn't simulate an iPhone or iPad in this situation. What Shairport Sync provides is an AirPlay destination for a Mac, iOS, AppleTV or HomePod.

If my guess is wrong, then you can install [`sps-alsa-explore`](https://github.com/mikebrady/sps-alsa-explore) to explore what audio devices may be present in your system that Shairport Sync can use.

Alternatively, you car stereo may have an analog auxiliary input ("AUX IN"). If so, then maybe the [CAR INSTALL.md](https://github.com/mikebrady/shairport-sync/blob/development/CAR%20INSTALL.md) guide might be relevant. BTW, please use the more up-to-date`development` version of the guide. However, you your steering-wheel controls won't work with it.

**Reply by jlodge10** [reply]
Thanks for the quick response. Yes your guesses are correct that it does play and display meta data over the cable as you stated. It does also have the ability to play from a mass storage device (such as a usb flash drive) which is why I was thinking Shairport Sync may work. Do you know of any other software that has the capability to do what I’m trying to do?

**Reply by mikebrady** [reply]
I'm afraid I don't, but some other readers may have suggestions.
